qiskit_experiments/characterization/t2ramsey.py

```diff
@@ -99,3 +102,203 @@ def circuits(self, backend: Optional[Backend] = None) -> List[QuantumCircuit]:
+        """
+        super().__init__([qubit])
+
+        default_options = self.experiment_options
+        default_options.delays = delays
+        default_options.unit = unit
+        default_options.offset_frequency = offset_frequency
+
+        self.set_experiment_options(**default_options.__dict__)
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        return Options(
+            delays=None,
+            unit="s",
+            offset_frequency=0.,
+            calibrations={
+                "sx": None,
+                "measure": None,
+            },
+        )
+
+    def circuits(self, backend: Optional[Backend] = None) -> List[QuantumCircuit]:
+        """Return a list of experiment circuits.
+
+        Args:
+            backend: Optional, a backend object.
+
+        Returns:
+            A list of :class:`QuantumCircuit`.
+        """
+        options = self.experiment_options
+
+        p_delay = Parameter("delay")
+
+        ram_x = QuantumCircuit(1)
+        ram_x.sx(0)
+        ram_x.delay(p_delay, 0, unit=options.unit)
+        ram_x.sx(0)
+        ram_x.measure_active()
+
+        ram_y = QuantumCircuit(1)
+        ram_y.sx(0)
+        ram_y.delay(p_delay, 0, unit=options.unit)
+        ram_y.rz(-np.pi/2, 0)
+        ram_y.sx(0)
+        ram_y.measure_active()
+
+        circs = []
+        for delay in options.delays:
+
+            # format delay to SI unit for analysis
+            if options.unit == "dt":
+                xval = delay * backend.configuration().dt
+            else:
+                if options.unit == "s":
+                    xval = delay
+                else:
+                    xval = apply_prefix(delay, options.unit)
+
+            metadata = {
+                "experiment_type": self._type,
+                "qubits": (self.physical_qubits[0], ),
+                "delay": delay,
+                "unit": options.unit,
+                "xval": xval,
+            }
+
+            # create ramsey x
+            assigned_x = ram_x.assign_parameters({p_delay: delay}, inplace=False)
+            assigned_x.metadata = metadata.copy()
+            assigned_x.metadata["post_pulse"] = "x"
+
+            # create ramsey y
+            assigned_y = ram_y.assign_parameters({p_delay: delay}, inplace=False)
+            assigned_y.metadata = metadata.copy()
+            assigned_y.metadata["post_pulse"] = "y"
+
+            circs.extend([assigned_x, assigned_y])
+
+        return circs
+
+    def _postprocess_transpiled_circuits(self, circuits, backend, **run_options):
+        """Attach gate calibration if available.
+
+        TODO need discussion,
+        - Ramsey should be able to run without calibration, same for T1 T2
+        - However we should be able to add calibration to any gate
+        """
+        for circ in circuits:
+            apply_calibration(self.experiment_options.calibrations, circ)
```

Comment:
We could be more reusable here with a transpiler pass. The problem is the same for running an algorithm, e.g. a VQE, and wanting to attach the calibrations. This feels more like the job for a transpiler pass otherwise we will duplicate this logic a lot. Note that the RamseyXY also brings in the problem of using qubit frequency values obtained from the calibrations. E.g. how does one use the spectroscopy frequencies and refine them with RamseyXY?

Reply:
Yes, I agree. We can move this to transpiler but we need to wait until next terra release. This is a temporary code for now.